hw/ip/otbn/dv/tracer/cpp/log_trace_listener.cc

```diff
@@ -0,0 +1,63 @@
+// Copyright lowRISC contributors.
+// Licensed under the Apache License, Version 2.0, see LICENSE for details.
+// SPDX-License-Identifier: Apache-2.0
+
+#include <cassert>
+#include <iomanip>
+#include <ios>
+#include <sstream>
+#include <string>
+
+#include "log_trace_listener.h"
+
+LogTraceListener::LogTraceListener(const std::string &log_filename)
+    : trace_log(log_filename, std::fstream::out) {
+  if (!trace_log.is_open()) {
+    std::ostringstream oss;
+    oss << "Could not open log file: " << log_filename;
+    throw std::runtime_error(oss.str());
+  }
+}
+
+void LogTraceListener::AcceptTraceString(const std::string &trace,
+                                         unsigned int cycle_count) {
+  assert(trace_log.is_open());
+
+  // Split the trace up into a vector of strings, one per line
+  auto trace_lines = SplitTraceLines(trace);
+  bool first_line = true;
+
+  // Write out the lines from the trace
+  for (auto &line : trace_lines) {
+    if (first_line) {
+      // Special handling for the first line
+      assert(line.size() > 1);
```

Comment:
I don't think this should be an assertion: bad SV code could cause it to fail, and it's probably cleaner to do some minimal error handling across the boundary. Maybe just spit out an "oh no!" line in this case?

Reply:
Output the error into the trace, did consider std::cerr but felt users of the class may not want it fiddling with their output streams.

Guess I could throw an exception but that looks rather like the assert, just allowing more elegant error messages on exit.